### PR TITLE
LGTM fixes for scfgrad

### PR DESCRIPTION
**Pull request by alenaizan**
## Description
Fix LGTM warnings for scfgrad

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/scfgrad/jk_grad.cc [hunk 1]
```diff
@@ -390,7 +390,7 @@ void DFJKGrad::build_Amn_terms()
         // > Integrals < //
         int nthread_df = df_ints_num_threads_;
 #pragma omp parallel for schedule(dynamic) num_threads(nthread_df)
-        for (long int PMN = 0L; PMN < NP * npairs; PMN++) {
+        for (long int PMN = 0L; PMN < (long) NP * npairs; PMN++) {
```
Comment (dgasmith) [inline psi4/src/psi4/scfgrad/jk_grad.cc]: Did we want to migrate all of these to `static_cast<long>(NP)` like the others?

Comment (loriab) [reply psi4/src/psi4/scfgrad/jk_grad.cc]: for new casts, yes, I agree.

File: psi4/src/psi4/scfgrad/response.cc [hunk 2]
```diff
@@ -125,16 +125,16 @@ std::shared_ptr<Matrix> RSCFDeriv::hessian_response()
...
-            psio_->write(PSIF_HESS,"Sai^A",(char*)Saip[0],nvir * nocc * sizeof(double),next_Sai,&next_Sai);
+            psio_->write(PSIF_HESS,"Sai^A",(char*)Saip[0],(unsigned long) nvir * nocc * sizeof(double),next_Sai,&next_Sai);
         }
         for (int A = 0; A < 3*natom; A++) {
-            psio_->write(PSIF_HESS,"Sij^A",(char*)Sijp[0],nocc * nocc * sizeof(double),next_Sij,&next_Sij);
+            psio_->write(PSIF_HESS,"Sij^A",(char*)Sijp[0],(unsigned long) nocc * nocc * sizeof(double),next_Sij,&next_Sij);
         }
         for (int A = 0; A < 3*natom; A++) {
-            psio_->write(PSIF_HESS,"Spi^A",(char*)Spip[0],nmo * nocc * sizeof(double),next_Spi,&next_Spi);
+            psio_->write(PSIF_HESS,"Spi^A",(char*)Spip[0],(unsigned long) nmo * nocc * sizeof(double),next_Spi,&next_Spi);
```
Comment (loriab) [inline psi4/src/psi4/scfgrad/response.cc]: and the `unsigned long`s can be `size_t`, I think

File: psi4/src/psi4/scfgrad/jk_grad.cc (hunk 1 above)
Comment (alenaizan) [reply psi4/src/psi4/scfgrad/jk_grad.cc]: updated.

File: psi4/src/psi4/scfgrad/response.cc (hunk 2 above)
Comment (alenaizan) [reply psi4/src/psi4/scfgrad/response.cc]: updated to `size_t`

**Pull request merged**